sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/ContainerRegistryCredentials.cs

```diff
@@ -280,22 +280,26 @@ private string GetScopeFromHeaders(HttpHeaders headers)
                 if (headerKVP.Key.ToLower() == challengeHeader)
                 {
                     headerValue = string.Join(",", headerKVP.Value);
+                    
                     break;
                 }
             }
-            
-            foreach (string part in headerValue.Split(','))
+
+            string scope = "";
+            int position = headerValue.IndexOf("scope=");
+            if (position > 0)
             {
-                string[] keyValues = part.Split(new char[] { '=' }, 2);
-                if (keyValues.Length != 2)
-                {
-                    throw new Exception($"{challengeHeader} has incorrect format, " +
-                        $"header key-value pair '{part}' does not have a value but in '{headerValue}'");
-                }
-                if (keyValues[0].ToLower().Trim() == "scope")
-                {
-                    return TrimDoubleQuotes(keyValues[1]);
-                } 
+                scope = headerValue.Substring(position);
```

Comment:
While this will likely work in most cases, it's not guaranteed that the server will always return a response in this format. Here's an example response from the server for the `WWW-Authenticate` header:
`Www-Authenticate: Bearer realm="https://myregistry.azurecr-test.io/oauth2/token",service="myregistry.azurecr-test.io",scope="foo:bar:baz"`
To generalize, this value's format is:
`<scheme> realm="<realm>", ..`
as shown here: https://docs.docker.com/registry/spec/api/#base
So while taking a substring, you should only read until the next `,` or if not present, the end of the string.